### PR TITLE
Piece features

### PR DESCRIPTION
**Pull request by bdistin**
### Description of the PR
Adds some new features to some old pieces.

### Changes Proposed in this Pull Request (List new items in CHANGELOG.MD)

- Adds ability for all Monitors to also act on message edits
- Adds static methods/properties to Extendables
- Makes Extendables consistent with all other pieces (+TS extends friendly)
- Adds once Events
- Adds Events for any Emitter
- Adds multiple Events per event (but still you really shouldn't)
- Makes onceReady an editable once event, without needing to edit KlasaClient
- Adds a catchUp option to ScheduledTasks (Schedule#create())

### Semver Classification

- [ ] This PR only includes documentation or non-code changes.
- [ ] This PR fixes a bug and does not change the (intended) framework interface.
- [x] This PR adds methods or properties to the framework interface.
- [x] This PR removes or renames methods or properties in the framework interface.

### CODE REVIEW

File: src/lib/structures/Event.js [hunk 1]
```diff
@@ -10,9 +10,29 @@ class Event extends Piece {
...
 	/**
 	 * @typedef {PieceOptions} EventOptions
+	 * @property {boolean} once If this event should only be run once and then unloaded
 	 * @memberof Event
 	 */
 
+	/**
+	 * @since 0.0.1
+	 * @param {KlasaClient} client The Klasa client
+	 * @param {MonitorStore} store The Monitor Store
```
Comment (UnseenFaith) [inline src/lib/structures/Event.js]: Copypasta is strong within this one

File: src/lib/structures/Event.js [hunk 2]
```diff
@@ -10,9 +10,29 @@ class Event extends Piece {
...
+	 * @param {string} file The path from the pieces folder to the monitor file
+	 * @param {boolean} core If the piece is in the core directory or not
+	 * @param {MonitorOptions} [options={}] Optional Monitor settings
```
Comment (UnseenFaith) [inline src/lib/structures/Event.js]: ditto

**Comment by kyranet** [comment]
The PR looks good to me, and ready for merge, but the `README.md` file is not updated to reflect the changes from this PR.

**Comment by bdistin** [comment]
Just need typings to be updated completely.

**Pull request merged**